File: contain_with_most_water/maxArea.c
int maxArea(int* height, int heightSize) 
{
	int i = 0,j = heightSize - 1;
	int ret = 0, tmp = 0;
	if (heightSize <= 1)
	{
		return 0;
	}
	while (i != j)
	{	
		if (height[i] < height[j])
		{
			tmp = (j-i)*height[i++];
		}
		else
		{
			tmp = (j-i)*height[j--];
		}
		ret = tmp > ret?tmp:ret;
	}
	return ret;  
}

File: contain_with_most_water/maxArea.java
class Solution {
    public int maxArea(int[] height) 
    {
        int left = 0, right = height.length - 1;
        int ret = 0;
        while (left != right)
        {
            ret = Math.max(ret, (right - left)*Math.min(height[right], height[left]));
            if (height[left] < height[right])
            {
                ++left;
            }
            else
            {
                --right;
            }
        }
        return ret;
    }
}

